ui/src/app/App.tsx

```diff
@@ -51,6 +52,8 @@ export const App = (): JSX.Element => {
   const completedUserIntro = useCompletedUserIntro();
   const debug = process.env.NODE_ENV === "development";
 
+  const appRoutes = useRoutes(routes);
```

Comment:
Is there a reason why you used `useRoutes` over regular JSX? The approach you're proposing would also work with maintaining the current JSX style of declaring routes.